src/Microsoft.AspNetCore.SignalR.Client.Core/HubConnection.Log.cs

```diff
@@ -174,6 +174,18 @@ private static class Log
             private static readonly Action<ILogger, string, Exception> _removingHandlers =
                LoggerMessage.Define<string>(LogLevel.Debug, new EventId(58, "RemovingHandlers"), "Removing handlers for client method '{MethodName}'.");
 
+            private static readonly Action<ILogger, string, Exception> _sendingMessageGeneric =
+                LoggerMessage.Define<string>(LogLevel.Debug, new EventId(59, "SendingMessageGeneric"), "Sending {MessageType} message.");
+
+            private static readonly Action<ILogger, string, Exception> _messageSentGeneric =
+                LoggerMessage.Define<string>(LogLevel.Debug, new EventId(60, "MessageSentGeneric"), "Sending {MessageType} message completed.");
+
+            private static readonly Action<ILogger, Exception> _acquiredConnectionLockForPing =
+                LoggerMessage.Define(LogLevel.Trace, new EventId(61, "AcquiredConnectionLockForPing"), "Have the connection lock, about to ping the server.");
+
+            private static readonly Action<ILogger, Exception> _unableToAcquireConnectionLockForPing =
+                LoggerMessage.Define(LogLevel.Trace, new EventId(62, "UnableToAcquireConnectionLockForPing"), "Tried to get connection lock, but it's in use. No need to ping because another message is being sent.");
```

Comment:
nit: EventName: "SkipPing" Message: "Skipping ping because a send is already in progress."